doc/design.rst

```diff
@@ -606,3 +597,61 @@ are deleted, the particular snapshot vanished and all snapshots
 depending on data that has been added in the snapshot cannot be restored
 completely. Restic is not designed to detect this attack.
 
+With the aforementioned assumptions and guarantees in mind, the following  are
+examples of things an adversary could achieve in various circumstances.
+
+An adversary with access to your backup storage location could: 
+- Delete or manipulate your backups, thereby impairing your ability to restore
+files from the compromised storage location. - Attempt a brute force password 
+guessing attack against a copy of the  repository (even more reason to use 
+long, random passwords).
+
+An adversary with write access to your files at the storage location could: 
+- Delete specific files (by using timestmaps to determine which files belong to
+what snapshots), resulting in snapshots vanishing and all dependent snapshots
+from being completely restored.
+
+An adversary who can passively monitor all Internet traffic could: 
+- Determine from where you create your backups (i.e., the location where the 
+requests originate).
+- Determine where you are storing your remote backups via HTTP headers (URLs)
+if you use a non-encrypted protocol. In nearly all cases, this will not be an
+issue because HTTP header sniffing is mitigated by HTTPS and most, if not all,
+popular HTTP-based cloud storage backends (e.g., S3, B2) require HTTPS. 
+Additionally, many non-HTTP protocols (e.g., SSH) are also secured. The REST 
+server is one case where unsecured HTTP could be used. However, system 
+administrators have the option (and are encouraged) to configure the REST 
+server to use HTTPS.
+
+An adversary who compromises (via malware, physical access, etc.) the  host
+system making backups could: 
+- Render the entire backup process untrustworthy (e.g., intercept password, 
+copy files, manipulate data).
+
+An adversary with the ability to add data to the repo (but not remove something)
```

Comment:
The adversary in this section is a very powerful one: It can add files and possesses the encryption password. In other word it is indistinguishable from a regular backup client with append-only access (via rclone with `--append-only` flag). This is this is very similar to the previous section, just with an added append-only restriction.

An attacker could also prevent restic from working by adding properly authenticated garbage to the repository. This is probably the most effective short-term attack for an attacker with append-only access.

Reply:
Yes, this is a very powerful one. Like you mentioned, the main difference is related to `--append-only`. 

@MichaelEischer, for your point about adding properly authenticated garbage to the repo, how big of a deal is it? 
- Would it corrupt the entire repo or only the adversary's own backed up files? As I understand it, the password a user uses to write data to the repo is different than the actual encryption password (the master key). 
- If [different repo keys](https://restic.readthedocs.io/en/stable/070_encryption.html#manage-repository-keys) are used by each user, does this make a difference? I use Restic in a single user scenario so don't fully understand the implications of multiple keys (passwords) beyond knowing it allows a password to be changed without having to re-encrypt all data (i.e., a key can exist for each file). 
- Do the integrity protections ([ciphertext verification](https://restic.readthedocs.io/en/stable/045_working_with_repos.html#checking-a-repo-s-integrity-and-consistency)) at time of restore come into play?

Reply:
As I'm looking into this a bit more, I understand a user's password would allow them to decrypt the repo's master key and that in turn would allow it to read the entire repository without needing a password at all (provided the user modifies to client code to support this). I didn't catch that at first and now see what you mean by possession the encryption password (at first I was thinking password / user key). This does further underscore how powerful this adversary is. 

I guess my main question is, what are you proposing here? Are you suggestion this entire section be removed because `append-only` limitation in this scenario isn't too different from an attacker's standpoint (i.e., it renders the entire backup process untrustworthy like the prior section)?